utils/ANLEra_special_macros.cfg

```diff
@@ -139,51 +146,36 @@
                 #textdomain wesnoth-ANLEra
             [/research]
 
-            [leader_option_1]
-                progress=0
-                target=4
-            [/leader_option_1]
+            {SET_SIDE_RESEARCH leader_option_1 0 4}
 
             ally_1=no
             ally_2=no
             ally_3=no
             ally_4=no
```

Comment:
@sevu, do you think it’s possible to (easily?) change these ally_[X] variables into something more flexible? Ideally something like a hash table from other languages (`ally[$unit.id]` / `ally["Dwarvish Fighter"]`) would solve a lot of issues.

Reply:
$ , . | {} [] = are all forbidden characters in variable names, so your exact examples would not work, but perhaps there is a better way of setting them up. I will look into this.